### ISSUE TITLE
CakeBuildLog interprets curly braces differently

### ISSUE THREAD

**Issue by agc93** [cake-build/cake#1257]
**This may be by design, so no problem if it is, happy to close.**
### What You Are Seeing?

Using double curly braces (i.e. `{{` or `}}`) results in double braces being output to the log.
### What is Expected?

Going by the most common parsing behaviour (i.e. `string.Format` and C# 6 `$` notation), this would result in a single `{` brace.

That is:

`Information("Sample with {{double}} braces");` `== "Sample with {{double}} braces"`
`Console.WriteLine($"Sample with {{double}} braces");` `== "Sample with {double} braces"`
### What version of Cake are you using?

0.16.1

---

This is caused by [this line](https://github.com/cake-build/cake/blob/develop/src/Cake/Diagnostics/Formatting/FormatParser.cs#L48) in the `FormatParser` used by `CakeBuildLog`. If this returned a literal token for `"{"` rather than `"{{"`, it would behave similarly to `string.Format`, and retain it's current property highlighting.

Happy to stick with the existing logic if this is intended behaviour, just caused a bit of an issue with Cake.Powershell which mixed `string.Format` invocations (from `ProcessArgumentBuilder`) with `ICakeLog` invocations, which each treat braces differently.

Let me know if this isn't clear.

**Comment by patriksvensson** [comment]
@agc93 Your issue has a point. If we could fix this and keep backwards compatibility it would be great.

**Comment by agc93** [comment]
Awesome, I'll try and take a look at this again once I get through some of the outstanding VS issues etc, unless someone else gets to it first :smile:

**Comment by devlead** [comment]
@agc93 excellent! I'll assign the issue to you.

**Comment by devlead** [comment]
@agc93 think this is sorted now, could you check and close if this now works as you expect.

**Comment by augustoproiete** [comment]
Marking this as resolved in milestone/release v0.30.0

```csharp
Information("Sample with {{double}} braces");
Console.WriteLine("Sample with {{double}} braces");
```

![image](https://user-images.githubusercontent.com/177608/108143758-b3e1de80-709e-11eb-9e83-2d4e4c746624.png)

---

Behavior remains the same with Cake v1.0.0

![image](https://user-images.githubusercontent.com/177608/108143868-e4c21380-709e-11eb-84e9-140690634e6b.png)
